packages/jellyfish-api-core/__tests__/category/account/getPendingFutureSwaps.test.ts

```diff
@@ -0,0 +1,317 @@
+
+    await testing.token.mint({
+      symbol: 'TSLA',
+      amount: 4
+    })
+    await testing.generate(1)
+
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { 'v0/params/dfip2203/active': 'false' } })
+    await testing.generate(1)
+
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { 'v0/params/dfip2203/reward_pct': futureRewardPercentage.toString(), 'v0/params/dfip2203/block_period': futureInterval.toString() } })
+    await testing.generate(1)
+
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { 'v0/params/dfip2203/active': 'true' } })
+    await testing.generate(1)
+
+    tslaAddress = await testing.generateAddress()
+
+    await testing.rpc.account.accountToAccount(collateralAddress, {
+      [tslaAddress]: '4@DUSD'
+    })
+    await testing.generate(1)
+
+    await testing.rpc.account.accountToAccount(collateralAddress, {
+      [tslaAddress]: '4@TSLA'
+    })
+    await testing.generate(1)
+  }
+
+  async function getNextSettleBlock (): Promise<number> {
+    const blockCount = await testing.rpc.blockchain.getBlockCount()
+    return blockCount + futureInterval - (blockCount % futureInterval)
+  }
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+    await setup()
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  describe('Single futureswap', () => {
+    it('Should getPendingFutureSwaps if futureswap TSLA for DUSD', async () => {
+      // Call getpendingfutureswaps before performing futureswap
+      {
+        const pendingFutureSwaps = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+        expect(pendingFutureSwaps).toStrictEqual({
+          owner: tslaAddress,
+          values: []
+        })
+      }
+
+      // futureswap
+      await testing.container.call('futureswap', [tslaAddress, '1@TSLA'])
+      await testing.generate(1)
+
+      // Call getpendingfutureswaps after performing futureswap
+      {
+        const pendingFutureSwaps = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+        expect(pendingFutureSwaps).toStrictEqual({
+          owner: tslaAddress,
+          values: [
+            {
+              source: '1.00000000@TSLA',
+              destination: 'DUSD'
+            }
+          ]
+        })
+      }
+
+      // Wait for next settle block
+      const nextSettleBlock = await getNextSettleBlock()
+      await testing.generate(nextSettleBlock - await testing.rpc.blockchain.getBlockCount())
+
+      // Call getpendingfutureswaps after futureswap is settled
+      {
+        const pendingFutureSwaps = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+        expect(pendingFutureSwaps).toStrictEqual({
+          owner: tslaAddress,
+          values: []
+        })
+      }
+    })
+
+    it('Should getPendingFutureSwaps if futureswap DUSD for TSLA', async () => {
+      // Call getpendingfutureswaps before performing futureswap
```

Comment:
It looks redundant as Line 139 does the same thing.
However, I believe every code in every Jest Item (It) should be independent.